flink-core/src/test/java/org/apache/flink/util/FileUtilsTest.java

```diff
@@ -265,6 +265,18 @@ public void testCompression() throws IOException {
 		FileUtils.expandDirectory(zip, new Path(extractDir.toAbsolutePath().toString()));
 
 		assertDirEquals(compressDir.resolve(originalDir), extractDir.resolve(originalDir));
+
+		FileUtils.deleteDirectory(extractDir.resolve(originalDir).toFile());
+		Files.delete(Paths.get(compressDir.resolve(originalDir) + ".zip"));
+
+		final java.nio.file.Path relativeCompressDir =
+			Paths.get(new File("").getAbsolutePath()).relativize(compressDir);
+		final Path relativeDirectoryZip = FileUtils.compressDirectory(
+			new Path(relativeCompressDir.resolve(originalDir).toString()),
+			new Path(compressDir.resolve(originalDir) + ".zip"));
+
+		FileUtils.expandDirectory(relativeDirectoryZip, new Path(extractDir.toAbsolutePath().toString()));
+		assertDirEquals(relativeCompressDir.resolve(originalDir), extractDir.resolve(originalDir));
```

Comment:
Please add a separate test case (`testCompressionOnRelativePath`?) and factor out common code to a private method.